### ISSUE TITLE
Add model evaluation report persistence

### ISSUE THREAD

**Issue by gelluisaac** [Gellu06/Nexora-NFT-Marketplace#329]
> 🔴 **Difficulty:** HARD · **Area:** ML · **Estimated effort:** 1–2 weeks

## 📌 Summary

Persist reproducible evaluation metadata alongside model metrics.

## 🎯 Context & motivation

Metrics without dataset, code, and configuration metadata are difficult to compare or audit.

## 🛠 Scope of work

1. Define an evaluation report schema.
2. Record dataset and configuration identifiers.
3. Persist report metadata with the metrics output.

## 📁 Files to touch

| Path | Change |
| --- | --- |
| `ml` | Implement evaluation report generation. |
| `packages/chenai-mlflow/src/index.ts` | Expose report metadata to consumers. |
| `ml/README.md` | Document the report format. |

_Paths are the expected shape of the change — if you need to deviate, explain why in the PR description._

## ✔️ Acceptance criteria

- [ ] Reports identify the model, data, and evaluation configuration.
- [ ] Reports are serializable and versioned.
- [ ] Sensitive training data is not copied into reports.

## 🧪 Testing requirements

- [ ] Test deterministic report serialization.
- [ ] Test missing metadata failures.
- [ ] Run package checks.

## ✅ CI requirements (non-negotiable)

**Every** check on your pull request must be green before review. Nothing is
merged on a red or skipped-because-broken pipeline.

| Pipeline | Runs when you touch | Must pass |
| --- | --- | --- |
| `Frontend CI` (`.github/workflows/frontend.yml`) | `apps/frontend/**` | install → `npm run lint` → `tsc --noEmit` → `npm run test` → `vite build` → artifact upload |
| `Backend CI` (`.github/workflows/backend.yml`) | `apps/backend/**`, `packages/**`, `pnpm-workspace.yaml` | install (pnpm) → `prisma generate` → `pnpm run lint` → `tsc -p tsconfig.json --noEmit` → `pnpm run test` → `pnpm run build`, plus the `Shared packages build` job |
| `Contracts CI` (`.github/workflows/contracts.yml`) | `contracts/**` | `cargo fmt --all --check` → `cargo clippy --workspace --all-targets` (with `RUSTFLAGS: -D warnings`, so **zero warnings**) → `cargo test --workspace` → `cargo build --workspace --release --target wasm32-unknown-unknown` |

Reproduce locally before pushing:

```bash
# frontend
cd apps/frontend && npm install && npm run lint && npx tsc --noEmit && npx vite build

# backend + shared packages
pnpm install && pnpm lint && pnpm build && pnpm test:all

# contracts
cd contracts && cargo fmt --all --check \
  && cargo clippy --workspace --all-targets -- -D warnings \
  && cargo test --workspace \
  && cargo build --workspace --release --target wasm32-unknown-unknown
```

Also required:

- [ ] Branch name follows `CONTRIBUTING.md` (`feat/…`, `fix/…`, `chore/…`, `docs/…`, `refactor/…`).
- [ ] Commits use Conventional Commits with the correct scope (e.g. `feat(backend): …`).
- [ ] `.github/PULL_REQUEST_TEMPLATE.md` filled in completely, screenshots for any UI change.
- [ ] At least **1 approving review**; branch is up to date with `main`.
- [ ] No generated artifacts, `.env` files, secrets, or lockfile churn unrelated to your change.

## 🧭 How to work on this

1. Comment on this issue to get it assigned — please don't open a PR for an
   unassigned issue.
2. Fork + branch from an up-to-date `main`.
3. Read `CONTRIBUTING.md` and `README.md` first; keep the change scoped to one
   domain (`apps/backend`, `apps/frontend`, `contracts`, `packages`, `ml`).
4. Keep the PR reviewable: if it grows past ~600 changed lines, split it and say
   so in the description.
5. Push early as a **draft PR** if you want directional feedback.

## 💬 Definition of done

PR merged into `main` with all acceptance criteria ticked, all CI pipelines green, docs updated where behaviour changed, and no follow-up work left undocumented (open a new issue for anything you deliberately defer).

**Comment by mysterio123865** [comment]
> Hi,
>
> I'd like to take on this issue.
>
> I've reviewed the evaluation report persistence task across `ml/`, `packages/chenai-mlflow/src/index.ts`, and `ml/README.md`. I understand the motivation precisely — metrics recorded without their accompanying dataset identifier, code version, and configuration snapshot are not reproducible and not auditable. Two runs producing different metric values are indistinguishable from each other unless the report captures exactly what data was evaluated, against which model version, and under which configuration. This implementation makes that context a first-class, versioned, serializable artifact persisted alongside the metrics output rather than an implicit assumption that exists only in the memory of whoever ran the evaluation.
>
> My implementation will cover the full scope of work across all three files.
>
> In `ml/` I will define an `EvaluationReport` schema as the foundational artifact. The schema will include a `schema_version` field for forward compatibility, a `model` block containing the model identifier and version hash, a `dataset` block containing the dataset identifier, split name, and a content hash of the dataset manifest — never a copy of the data itself, satisfying the acceptance criterion that sensitive training data is never copied into reports — an `evaluation_config` block containing all configuration parameters that affect metric values such as thresholds, batch size, and random seed, a `metrics` block containing the evaluation output values with their names and types, and a `generated_at` ISO 8601 timestamp. I will implement report generation as a dedicated module that populates this schema from the evaluation run context and serializes it to JSON. The serialization will be deterministic — fields emitted in a fixed canonical order — so that two reports generated from identical inputs produce byte-identical output, making report comparison and hashing reliable.
>
> In `packages/chenai-mlflow/src/index.ts` I will expose the `EvaluationReport` type and a `persistReport` function that consumers can call to write the report alongside the metrics output. The function signature will make all required fields explicit so that a missing model identifier, dataset identifier, or configuration block produces a TypeScript compile-time error rather than a silent runtime gap. I will add a JSDoc comment on every exported symbol linking to the report format documentation in `ml/README.md` so consumers can navigate directly from the package to the schema reference.
>
> In `ml/README.md` I will add an Evaluation Report Format section documenting the complete schema — every field with its type, whether it is required or optional, and what value is expected — alongside a complete synthetic example report containing no sensitive information, validated against the current implementation before submission.
>
> The test suite will cover three scenarios explicitly required by the acceptance criteria. For deterministic serialization I will generate two reports from identical inputs and assert byte-identical JSON output. For missing metadata failures I will assert that omitting each required field — model identifier, dataset identifier, configuration — produces the expected error at report construction time rather than silently emitting an incomplete report. For the package checks I will run the full typecheck against `packages/chenai-mlflow/src/index.ts` confirming the exposed types compile cleanly.
>
> Before pushing I will reproduce the full CI pipeline locally — frontend, backend and shared packages, and contracts — confirming all checks are green. The branch will follow the `feat/…` naming convention from `CONTRIBUTING.md`, commits will use Conventional Commits with the correct scope, and the PR template will be filled in completely with no generated artifacts, `.env` files, or unrelated lockfile churn included.
>
> Please assign this to me and I will begin immediately.

**Comment by Stanley-Owoh** [comment]
Good day maintainer, i would like to work on this issue. i will add a versioned evaluation report schema with model, dataset, and configuration metadata, expose it through `chenai-mlflow`, document the format, and add deterministic serialization and validation tests without storing sensitive training data.
